libraries/core_libs/consensus/src/transaction/transaction_manager.cpp

```diff
@@ -341,24 +337,48 @@ void TransactionManager::moveNonFinalizedTransactionsToTransactionsPool(std::uno
 }
 
 // Verify all block transactions are present
-bool TransactionManager::checkBlockTransactions(DagBlock const &blk) {
+std::optional<std::map<trx_hash_t, std::shared_ptr<Transaction>>> TransactionManager::getBlockTransactions(
```

Comment:
why it needs to be map? and does it need to be optional? can we return empty container?

Reply:
It is a map because later it is needed as a map to match the estimates within the dag block. I think it is cleaner to have an optional to point that it is an error and not just no transactions found. But either way is good